packages/block-editor/src/components/block-list/block-wrapper.js

```diff
@@ -216,6 +221,21 @@ const BlockComponent = forwardRef(
 				{ children }
 			</Animated>
 		);
+
+		if ( supportAlignments ) {
+			return (
+				<div
+					className={ classnames( 'wp-block wp-align-wrapper', {
+						[ `wp-align-${ attributes.align }` ]: !! attributes.align,
```

Comment:
Restating my concerns in https://github.com/WordPress/gutenberg/issues/20650#issuecomment-595807350, I think now is the prime time to revise the naming scheme for the alignment/position/width options. `wp-align-wide` shouldn't be a thing because "wide" is not really an alignment, and `wp-align-left` is misleading if it actually refers to a float.

Reply:
What do you propose? for me we should stay consistent in terms of classNames, in terms of labels in the UI it can be different

Reply:
If we were to put them all under a single naming scheme, I'd go with something like this:

`wp-position-xxx` with the `xxx` replaced with one of the following for each of the possible values:

- `align-left` (as in actual left alignment)
- `align-right`
- `align-center`
- `float-left` (or maybe `pull-left`)
- `float-right` (or maybe `pull-right`)
- `stretch-wide-width` (or maybe just `wide`)
- `stretch-full-width` (or maybe just `full-width`)

I'm not sure whether `wp-position-` is the right naming scheme, though. CSS already has a thing called `position` that isn't really the same concept. Perhaps it may also be worth distinguishing from potential vertical options by using `wp-horizontal-position-` or `wp-x-position-` or something like that.

But I'm not convinced that different options should use the same naming scheme if they're technically different things. If we were to use more diverse names depending on what exactly the position/alignment/whatever actually is, I would go with something like the following:

- `wp-align-left`
- `wp-align-right`
- `wp-align-center` (or possibly `wp-align-center-horizontally` or `wp-align-center-x` to distinguish from potential vertical centering or something like that in the future)
- `wp-float-left` (or maybe `wp-pull-left`)
- `wp-float-right` (or maybe `wp-pull-right`)
- `wp-width-wide`
- `wp-width-full`

I prefer this naming scheme since it's more concise, and doesn't try to fit a redundant, possibly incorrect word like "position" into the class names.

Reply:
I don't think we can change the name of the alignments today `left`/`right`/`wide`/`full`/`center` are  here to stay. I don't think it's worth the hassle to rename these for BC. We should try to figure out a prefix that work well for these. Otherwise, we'll just add complexity to map the alignment name with the classname without high benefits.

Also cc @aduth as my go to naming person :P

Reply:
@youknowriad `wp-align-xxx` is already different from the existing `alignxxx` class names, though, so we're already updating the class names, aren't we?

Reply:
Yes, but I'm talking about the attribute values, not the classNames.We also need to keep. BC for existing themes with `data-align[value]`  stylesheets. Again renaming this is possible but not without needless complexity.

Reply:
Can we omit the `wp-` prefix for alignments?

Reply:
@jasmussen I'm fine with removing them.

Reply:
@youknowriad Personally, I think that changing the attribute values, even if it means some somewhat complex back-compat, is well worth it to get class names that actually make sense. Changing how block alignments work is likely to break styles anyway, so we might as well take advantage of the opportunity so people don't have to update their styles twice.

Also, do we even need `data-align` anymore? See https://github.com/WordPress/gutenberg/pull/20689#discussion_r389636991.